src/unix/lwt_unix_unix.h

```diff
@@ -1826,12 +1826,21 @@ static void worker_readdir_n(struct job_readdir_n *job)
         /* An error happened. */
         if (entry == NULL && errno != 0) {
             job->error_code = errno;
+            job->count = i - 1;
             return;
         }
 
         /* End of directory reached */
         if (entry == NULL && errno == 0) break;
 
+        /* readdir is good */
+        char *name = strdup(entry->d_name);
+        if (name == NULL) {
+            job->error_code = errno;
```

Comment:
Good call here, I didn't even know that `strdup` actually sets `errno` (to `ENOMEM`) :p